src/main/java/com/softserve/teamproject/entity/Student.java

```diff
@@ -111,7 +111,7 @@ public boolean equals(Object otherObject) {
 
   @Override
   public int hashCode() {
-    return Objects.hash(firstName, lastName, group.getId());
+    return Objects.hash(firstName, lastName);
```

Comment:
well, if we have equals and hashcode of name and lastname of student, then two students with same names but in different groups will be equal, and you can't you HashSet etc.